modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/SelectTag.java

```diff
@@ -0,0 +1,185 @@
+
+import java.util.List;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class SelectTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		if (Validator.isNotNull(getLabel())) {
+			setDynamicAttribute(StringPool.BLANK, "aria-label", getLabel());
+		}
+
+		return super.doStartTag();
+	}
+
+	public String getLabel() {
+		return LanguageUtil.get(
+			TagResourceBundleUtil.getResourceBundle(pageContext), _label);
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public List<SelectOption> getOptions() {
+		return _options;
+	}
+
+	public boolean isDisabled() {
+		return _disabled;
+	}
+
+	public boolean isMultiple() {
+		return _multiple;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	public void setMultiple(boolean multiple) {
+		_multiple = multiple;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	public void setOptions(List<SelectOption> options) {
+		_options = options;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_disabled = false;
+		_label = null;
+		_multiple = false;
+		_name = null;
+		_options = null;
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("form-control");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		if (Validator.isNotNull(_label)) {
+			jspWriter.write("<label>");
+			jspWriter.write(_label);
+			jspWriter.write("</label>");
+		}
+
+		jspWriter.write("<select ");
+
+		super.writeDynamicAttributes();
+
+		super.writeCssClassAttribute();
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(getId())) {
+			super.writeIdAttribute();
+		}
+
+		if (_multiple) {
+			jspWriter.write(" multiple");
+		}
+
+		if (Validator.isNotNull(_name)) {
+			jspWriter.write(" name=\"");
+			jspWriter.write(_name);
+			jspWriter.write("\"");
+		}
+
+		jspWriter.write(">");
+
+		for (SelectOption option : _options) {
+			jspWriter.write("<option value=\"");
+			jspWriter.write(option.getValue());
+			jspWriter.write("\"");
+
+			if (option.isSelected()) {
+				jspWriter.write(" selected");
+			}
+
+			jspWriter.write(">");
+			jspWriter.write(option.getLabel());
+			jspWriter.write("</option>");
+		}
+
+		jspWriter.write("</select>");
+
+		return SKIP_BODY;
+	}
+
+	@Override
+	protected void writeCssClassAttribute() throws Exception {
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write(" class=\"form-group\"");
+	}
+
+	@Override
+	protected void writeDynamicAttributes() {
+	}
+
+	@Override
+	protected void writeIdAttribute() {
+	}
```

Comment:
@kresimir-coko just a doubt: is this intended? If we override these methods but leave them empty they won't do [this](https://github.com/liferay/liferay-portal/blob/2ba32f018d39b3eaacd32ff6c0fed7cfb806cff7/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/servlet/taglib/BaseContainerTag.java#L387-L404)

Reply:
I think it's intended because we don't want an `id` on the container element (`<div>`), but we want it on the `<select>`

Reply:
ok thanks for clarifying, I didn't see [that](https://github.com/liferay-frontend/liferay-portal/pull/492/files/b5cde87d570a421d1474bc560f43b4766fb96e4f#diff-40c9b154810e3ef867cb61c2c11ad2accd0159d0e36350c91a6e7bc734dc7292R128),  now it makes sense